src/writers/sink.creator.cpp

```diff
@@ -8,8 +8,11 @@
 namespace zarr = acquire::sink::zarr;
 namespace common = zarr::common;
 
-zarr::SinkCreator::SinkCreator(std::shared_ptr<common::ThreadPool> thread_pool_)
+zarr::SinkCreator::SinkCreator(
+  std::shared_ptr<common::ThreadPool> thread_pool_,
+  std::shared_ptr<common::S3ConnectionPool> connection_pool)
```

Comment:
What are the trade-offs between having a connection pool and having threads create their own connections?

Reply:
The threads aren't using their own connections. Each connection would be accessed by (or belong to, as an alternative) an `S3Sink` object, which roughly corresponds to an open connection to an S3 object (see #278 for usage). Because there's technically no bound on these, I felt it was better to explicitly cap the number of open connections by using a connection pool.

Reply:
Hmm, I don’t think I understand. Maybe we should discuss this over Zoom? Either way, I'm not going to block on this

Reply:
Check Slack